### PR TITLE
Typecheck fully resolved imports

### PR DESCRIPTION
**Pull request by TimWSpence**

### CODE REVIEW

File: tests/src/main/scala/org/dhallj/tests/acceptance/AcceptanceSuccessSuite.scala [hunk 1]
```diff
@@ -92,28 +96,35 @@ abstract class CachingExprOperationAcceptanceSuite(transformation: Expr => Expr)
...
-class HashingSuite(val base: String) extends ResolvingExprAcceptanceSuite[String] {
+class CachingTypeCheckingSuite(val base: String)
+    extends ExprOperationAcceptanceSuite(Expr.Util.typeCheck(_))
+    with CachedResolvingInput
+class TypeCheckingSuite(val base: String)
+    extends ExprOperationAcceptanceSuite(Expr.Util.typeCheck(_))
+    with ResolvingInput
+class AlphaNormalizationSuite(val base: String) extends ExprOperationAcceptanceSuite(_.alphaNormalize) with ParsingInput
+class NormalizationSuite(val base: String) extends ExprOperationAcceptanceSuite(_.normalize) with CachedResolvingInput
+class NormalizationUSuite(val base: String) extends ExprOperationAcceptanceSuite(_.normalize) with ParsingInput
```
Comment (travisbrown) [inline tests/src/main/scala/org/dhallj/tests/acceptance/AcceptanceSuccessSuite.scala:107]: I'm not sure about the `U` suffix here. What does it mean?

File: tests/src/test/scala/org/dhallj/tests/acceptance/AcceptanceSuites.scala [hunk 2]
```diff
@@ -25,6 +25,9 @@ class TypeCheckingSimpleSuite extends CachingTypeCheckingSuite("type-inference/s
 class TypeCheckingUnitSuite extends CachingTypeCheckingSuite("type-inference/success/unit")
 class TypeCheckingRegressionSuite extends TypeCheckingSuite("type-inference/success/regression")
 class TypeCheckingOtherSuite extends TypeCheckingSuite("type-inference/success") {
+  //TODO prelude is WAY too slow
```
Comment (travisbrown) [inline tests/src/test/scala/org/dhallj/tests/acceptance/AcceptanceSuites.scala:28]: What's way too slow? To be honest I'd rather hold off on this change for now and keep it easy to make sure we're correctly type-checking the unnormalized prelude in CI, etc.

File: tests/src/main/scala/org/dhallj/tests/acceptance/AcceptanceSuccessSuite.scala (hunk 1 above)
Comment (TimWSpence) [reply tests/src/main/scala/org/dhallj/tests/acceptance/AcceptanceSuccessSuite.scala:107]: It means I wanted to call it `Unit` but that name was already taken 😂

File: tests/src/test/scala/org/dhallj/tests/acceptance/HashingPreludeSuites.scala [hunk 3]
```diff
@@ -1,66 +1,66 @@
...
 
-class HashingPreludeBoolAndSuite extends HashingSuite("semantic-hash/success/prelude/Bool/and")
-class HashingPreludeBoolBuildSuite extends HashingSuite("semantic-hash/success/prelude/Bool/build")
-class HashingPreludeBoolEvenSuite extends HashingSuite("semantic-hash/success/prelude/Bool/even")
-class HashingPreludeBoolFoldSuite extends HashingSuite("semantic-hash/success/prelude/Bool/fold")
-class HashingPreludeBoolNotSuite extends HashingSuite("semantic-hash/success/prelude/Bool/not")
-class HashingPreludeBoolOddSuite extends HashingSuite("semantic-hash/success/prelude/Bool/odd")
-class HashingPreludeBoolOrSuite extends HashingSuite("semantic-hash/success/prelude/Bool/or")
-class HashingPreludeBoolShowSuite extends HashingSuite("semantic-hash/success/prelude/Bool/show")
+class HashingBoolAndSuite extends HashingSuite("semantic-hash/success/prelude/Bool/and")
```
Comment (travisbrown) [inline tests/src/test/scala/org/dhallj/tests/acceptance/HashingPreludeSuites.scala:3]: I think I'd like to keep it easy to distinguish the prelude tests from others (both in the results and with `testOnly`). I'm not necessarily against removing the `Prelude` from the test class name, but if we do I think these should go in a new `org.dhallj.tests.acceptance.prelude` package.

File: tests/src/test/scala/org/dhallj/tests/acceptance/AcceptanceSuites.scala (hunk 2 above)
Comment (TimWSpence) [reply tests/src/test/scala/org/dhallj/tests/acceptance/AcceptanceSuites.scala:28]: The prelude test is what was hanging and aborting the CI builds. We can do, it just means we don't conform to the spec and will have to disable one or two other tests and I thought most of this was covered in the `Prelude` suites anyway

File: tests/src/test/scala/org/dhallj/tests/acceptance/HashingPreludeSuites.scala (hunk 3 above)
Comment (TimWSpence) [reply tests/src/test/scala/org/dhallj/tests/acceptance/HashingPreludeSuites.scala:3]: Oh sorry!!! That was an unintentional `sed` change. Will revert

File: tests/src/test/scala/org/dhallj/tests/acceptance/AcceptanceSuites.scala (hunk 2 above)
Comment (travisbrown) [reply tests/src/test/scala/org/dhallj/tests/acceptance/AcceptanceSuites.scala:28]: Okay, let's go ahead and merge this after the `Prelude` name change revert and I'll take a closer look later at making sure we can run this.

**Comment by travisbrown** [comment]
@TimWSpence Thanks! Will merge on green.

File: tests/src/test/scala/org/dhallj/tests/acceptance/AcceptanceSuites.scala (hunk 2 above)
Comment (TimWSpence) [reply tests/src/test/scala/org/dhallj/tests/acceptance/AcceptanceSuites.scala:28]: 👍 I'm going to have a look into the Haskell implementation to see if it does any optimization (caching type-checking results, etc)

**Pull request merged**